atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -68,14 +75,26 @@
 
     private List<InetSocketAddress> cassandraHosts;
 
+    private Set<InetSocketAddress> localHosts;
```

Comment:
for thread safety, this really needs to be `volatile`. Let's briefly catch up about what can happen if you _don't_ do this (bluf: it's bad).